modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj

```diff
@@ -315,24 +324,13 @@
         [subclause]))
     subclauses)))
 
-;; From the dox:
-;;
-;; The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries, and common table
-;; expressions, unless TOP, OFFSET or FOR XML is also specified.
-;;
-;; To fix this we'll add a max-results LIMIT to the query when we add the order-by if there's no `limit` specified,
-;; but not for `top-level` queries (since it's not needed there)
 (defmethod sql.qp/apply-top-level-clause [:sqlserver :order-by]
   [driver _ honeysql-form {:keys [limit], :as query}]
   ;; similar to the way we optimize GROUP BY above, optimize temporal bucketing in the ORDER BY if possible, because
   ;; year(), month(), and day() can make use of indexes while DateFromParts() cannot.
   (let [query         (update query :order-by optimize-order-by-subclauses)
-        add-limit?    (and (not limit) (pos? sql.qp/*nested-query-level*))
-        honeysql-form ((get-method sql.qp/apply-top-level-clause [:sql-jdbc :order-by])
-                       driver :order-by honeysql-form query)]
-    (if-not add-limit?
-      honeysql-form
-      (sql.qp/apply-top-level-clause driver :limit honeysql-form (assoc query :limit qp.i/absolute-max-results)))))
+        parent-method (get-method sql.qp/apply-top-level-clause [:sql-jdbc :order-by])]
+    (parent-method driver :order-by honeysql-form query)))
```

Comment:
just as a simple question, i've always wondered about this `get-method` and then call the returned method pattern versus just `(sql.qp/apply-top-level-clause :sql-jdbc ...)` and calling the multimethod yourself. Is it just to prevent being sloppy and going into an infinite loop without changing the driver in the hierarchy?

Reply:
actually it's a pretty important difference. If `sql.qp/apply-top-level-clause` in turn calls another method (e.g. `sql.qp/->honeysql` then it's super important that it calls the `:sqlserver` version of that method and not the default `:sql` version. So it's super important you pass in the actual `driver` parameter you got in

Reply:
Ah. That's a subtlety i wouldn't have thought of but seems obvious after you mention it. Thanks